pkg/pulumiyaml/ast/expr.go

```diff
@@ -4,13 +4,16 @@ package ast
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 
 	"github.com/hashicorp/hcl/v2"
 	"github.com/pulumi/pulumi-yaml/pkg/pulumiyaml/syntax"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
 )
 
+const fnInvokeRegexPattern = "Fn::.+:.+(:.+)?"
```

Comment:
We try to avoid compiling precomputed regexes in functions to avoid recompiling them. This is idiomatic:
```suggestion
var fnInvokeRegex = regex.MustCompile("Fn::.+:.+(:.+)?")
```